gpjax/fit.py

```diff
@@ -33,7 +33,7 @@
 def fit(
```

Comment:
NB: should `fit()` always jit the code itself ? should we make an issue for that?

Reply:
I think `fit` should just accept a callable. Let's leave it as is for now and, once merged, we can open a PR to resolve #235